src/qiskit_qir/translate.py

```diff
@@ -2,64 +2,126 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 ##
-from qiskit_qir.elements import QiskitModule
 from qiskit_qir.visitor import BasicQisVisitor
 from qiskit.circuit.quantumcircuit import QuantumCircuit
+from typing import List, Tuple, Union
+from pyqir import Context, Module
+from qiskit_qir.elements import QiskitModule
 
 
 def to_qir(
-    circuit: QuantumCircuit, profile: str = "AdaptiveExecution", **kwargs
+    circuits: Union[QuantumCircuit, List[QuantumCircuit]],
+    profile: str = "AdaptiveExecution",
+    **kwargs
 ) -> str:
     r"""Converts the Qiskit QuantumCircuit to QIR as a string
 
-    :param circuit:
-        Qiskit circuit to be converted to QIR
-    :type circuit: ``QuantumCircuit``
+    :param circuits:
+        Qiskit circuit(s) to be converted to QIR
+    :type circuit: ``Union[QuantumCircuit, List[QuantumCircuit]]``
     :param profile:
         The target profile for capability verification
     :type profile: ``str``
     :param \**kwargs:
         See below
+    :returns: The LLVM IR string representation of the input.
 
     :Keyword Arguments:
         * *record_output* (``bool``) --
           Whether to record output calls for registers, default `True`
-        * *use_static_qubit_alloc* (``bool``) --
-          Whether to use static qubit allocation, default `True`
-        * *use_static_result_alloc* (``bool``) --
-          Whether to use static result allocation, default `True`
+        * *emit_barrier_calls* (``bool``) --
+          Whether to emit barrier calls in the QIR, default `False`
     """
-
-    module = QiskitModule.from_quantum_circuit(circuit=circuit)
-    visitor = BasicQisVisitor(profile, kwargs=kwargs)
-    module.accept(visitor)
-    return visitor.ir()
+    _, module = _build_module(circuits, profile, **kwargs)
+    return str(module)
 
 
 def to_qir_bitcode(
-    circuit: QuantumCircuit, profile: str = "AdaptiveExecution", **kwargs
+    circuits: Union[QuantumCircuit, List[QuantumCircuit]],
+    profile: str = "AdaptiveExecution",
+    **kwargs
 ) -> bytes:
     r"""Converts the Qiskit QuantumCircuit to QIR bitcode as bytes
 
-    :param circuit:
+    :param circuits:
+        Qiskit circuit to be converted to QIR
+    :type circuit: ``Union[QuantumCircuit, List[QuantumCircuit]]``
+    :param profile:
+        The target profile for capability verification
+    :type profile: ``str``
+    :param \**kwargs:
+        See below
+    :returns: The LLVM bitcode representation of the input.
+
+    :Keyword Arguments:
+        * *record_output* (``bool``) --
+          Whether to record output calls for registers, default `True`
+        * *emit_barrier_calls* (``bool``) --
+          Whether to emit barrier calls in the QIR, default `False`
+    """
+
+    _, module = _build_module(circuits, profile, **kwargs)
+    return module.bitcode
+
+
+def to_qir_bitcode_with_entry_points(
```

Comment:
I guess this exists only to avoid breaking to_qir_bitcode? I think the name is clunky because it's not really substantially different from to_qir_bitcode. Can you break the API?

Reply:
As an alternative, would it be reasonable to keep the existing `to_qir_bitcode` the same and add a function like `entry_points_from_bitcode` that takes the bc and returns the list of strings?

Reply:
@swernli I don't think we're guaranteed to get the exact order of entry points if we do two passes.

Reply:
ah, I see, the correlation implied by the ordering is important, I missed that detail.

Reply:
@samarsha I reworked the API to be `to_qir_module` letting users convert to bitcode or IR as they wish. This seemed to be the most flexible solution.